### ISSUE TITLE
Doc is not clear on how to expand snippet on keypress

### ISSUE THREAD

**Issue by joeitu** [occivink/kakoune-snippets#39]
Hello,
I think most of  people expect from snippet plug to have to following behavior:
type `for` then press a key so the `for` expand to 
```
for $X in $COL:
    $BODY
```
and then press another key  ( usually tab)   to move from $X to $COL to $BODY.

I think this the default way of using snippet, but it's really not clear how to do that in the doc. I feel this should be inside the `##Usage` paragraph

**Comment by occivink** [comment]
You're right, I think the documentation focuses a bit too much on internals as opposed to having examples. I'll try to improve this.
